nodi_backend/apps/datasets/views.py

```diff
@@ -129,18 +129,34 @@ def search(self, request):
         datatypes = request.query_params.get('datatypes', [])
         categories = request.query_params.get('categories', [])
 
+        #filters based on name query
```

Comment:
Because this logic is not integral to the success of the search functionality, you should wrap it in a try catch, such that if it fails for reasons you missed, it doesn't affect the entire method.